packages/api-codec/src/ExtrinsicIndex.ts

```diff
@@ -0,0 +1,8 @@
+// Copyright 2017-2018 @polkadot/api-codec authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import U16 from './U16';
+
+export default class ExtrinsicIndex extends U16 {
```

Comment:
Is ExtrinsicIndex U16 or U32? From old tests it seems like U16, but [here](https://github.com/paritytech/substrate/search?utf8=%E2%9C%93&q=extrinsic_Index&type=) it seems U32.

Reply:
Seems like u32 based on what you linked. The issue with the old tests are that the length encoding is now completely different, so there are some hoops to jump through to get it going or converted - so e2e is probably easier (initial) tests.

Basically, say you had a Bytes of 16 bytes in length (`0123456789abcdef`), previously it would have been

```
[
// length
16, 0, 0, 0,
// bytes
0, 1, 2, 3, 4, 5, 6, 7, 8, 9, a, b, c, d, e, f
]
```

now

```
[
// length (encoded with Compact
16 << 2,
// bytes
0, 1, 2, 3, 4, 5, 6, 7, 8, 9, a, b, c, d, e, f
]
```

(and then when it is 2 bytes or 4 bytes, it gets weirder...)